webapp/src/components/training/Trainer.vue

```diff
@@ -171,17 +217,23 @@ async function startTraining(distributed: boolean): Promise<void> {
 
   const disco = await Disco.fromTask(props.task, CONFIG.serverUrl, {
     logger: {
-      success: (msg: string) => {
-        messages.value = messages.value.push(msg);
-      },
-      error: (msg: string) => {
-        messages.value = messages.value.push(msg);
-      },
+      success: (msg: string) => messages.value = messages.value.push(msg),
+      error: (msg: string) => messages.value = messages.value.push(msg),
+      setStatus: (status: TrainingStatus) => trainingStatus.value = status
     },
     memory: memoryStore.useIndexedDB ? new IndexedDB() : new EmptyMemory(),
-    scheme: distributed ? props.task.trainingInformation.scheme : "local",
+    scheme: isTrainingAlone.value ? "local": props.task.trainingInformation.scheme,
   });
 
+  // Store the cleanup function such that it can be ran if users
+  // manually interrupt the training
+  cleanupTrainingSessionFn.value = async () => {
```

Comment:
ho, that's for capturing `disco` innit. hum, tricky, a bit hard to comprehend. consider moving the other cleanups (`displayModelCaching`, `*training*`, …) to `cleanupTrainingSession` and update the name to smth like `cleanupDisco` or `cleanupServerConnection`.

Reply:
That's the fix I found to ensure we still called the cleanup event when the stop error was not caught, maybe it's not useful anymore with the changes in your PR

Reply:
I'm not sure if I misunderstood or if there is a typo but the other cleanups like displayModelCaching, isTraining etc are already included in cleanupTrainingSessionFn, is this what you meant?

Reply:
Follow up: I had to make sure to write `cleanupDisco.value = async () => await disco.close()` rather than `cleanupDisco.value = disco.close` otherwise `this` was undefined when trying to access it within the `disco.close()` method